### PR TITLE
Parallelise import of COSMIC variants

### PR DESCRIPTION
**Pull request by nuno-agostinho**
[ENSVAR-4863](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4863)

This update allows to import COSMIC variants in parallel by following these steps:

1. Chunk the COSMIC CSV file into one file per chromosome (25 files):
```bash
awk -F\| '{print>"input/cosmic_b38_v96_"$1".csv"}' input/cosmic_b38_v96.csv
```

2. Remove previous COSMIC entries:
```bash
perl ensembl-variation/scripts/import/remove_cosmic_or_hgmd.pl --registry ensembl.registry --source COSMIC
```

3. Import COSMIC variants in parallel for each file, creating temporary tables with the current index as their suffix (e.g. for index 2, the tables `MTMP_tmp_cosmic2`, `MTMP_tmp_cosmic_phenotype2` and `MTMP_tmp_cosmic_synonym2` are created):
```bash
for chr in {1..25}; do
  echo $chr
  perl ensembl-variation/scripts/import/import_cosmic.pl \
        --registry ensembl.registry \
        --import input/cosmic_head_$chr.csv \
        --version 96 \
        --index $chr
done
```

4. Clean-up to join the information of temporary tables into tables `MTMP_tmp_cosmic`, `MTMP_tmp_cosmic_phenotype` and `MTMP_tmp_cosmic_synonym` and to delete the previous temporary tables:
```bash
import_cosmic.pl --registry ensembl.registry --cleanup
```

The instructions here will be added to our Confluence documentation.

## Testing

Copy the directory mentioned in the last comment from [ENSVAR-4863](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4863):
* It already has the input files chunked ready to go.
* You can run scripts `remove_cosmic.sh`, `parallel_import_cosmic.sh` and `cleanup_cosmic.sh` to perform the steps mentioned above.
* The scripts target the database `nuno_homo_sapiens_variation_108_38_parallel_cosmic` @ var-prod-3.

### CODE REVIEW

**Comment by dglemos** [comment]
Now that we have branched for release 109 this PR needs to be done against postreleasefix/109

**Comment by dglemos** [comment]
@nuno-agostinho there is one update missing in this PR:
[This commit](https://github.com/Ensembl/ensembl-variation/commit/0d5ec5d5350d62928e0736f96e9b851f1a9e0749) fixes a bug in ClinVar - we need the same update for COSMIC.

File: scripts/import/import_cosmic.pl [hunk 1]
```diff
@@ -358,27 +441,141 @@ sub get_variation_set_id {
...
+    if (defined $data_tv->[0]->[0]) {
+      my $update_vf_sth = $dba->dbc()->prepare($update_vf_smt);
+      # warn "Inserting variation_feature: $data_tv->[0]->[0], $data_tv->[0]->[1]\n";
+      $update_vf_sth->execute($data_tv->[0]->[1], $data_tv->[0]->[0]) || die "Error updating consequence_types in table variation_feature\n";
+    }
+  }
+
+  # Update variation_set in variation_feature table
+  my $vf_set_upd_sth = $dbh->prepare(qq[ UPDATE variation_feature
+                                         SET variation_set_id = '$variation_set_cosmic,$variation_set_pheno'
```
Comment (dglemos) [inline scripts/import/import_cosmic.pl:572]: Can you please fix one of my mistakes? This query is updating correct the variation_set_id, it should be:
`SET variation_set_id = concat($variation_set_cosmic,',', $variation_set_pheno)`

Comment (nuno-agostinho) [reply scripts/import/import_cosmic.pl]: Done!

**Review by dglemos — APPROVED** [review]
Thanks for the updates! The script works, and the new options cleanup and index work as expected.

**Comment by dglemos** [comment]
Merged into release/109 and main

**Pull request merged**